src/sentry/replays/lib/event_linking.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+from hashlib import md5
+from typing import TYPE_CHECKING, TypedDict
+
+from sentry.utils import json
+
+if TYPE_CHECKING:
+    from sentry.eventstore.models import BaseEvent
+
+
+class EventLinkKafkaMessage(TypedDict):
+    type: str
+    start_time: int
+    replay_id: str
+    project_id: int
+    segment_id: None
+    payload: list[int]
+    retention_days: int
+
+
+class EventLinkPayload(TypedDict):
+    type: str
+    replay_id: str
+    error_id: str
+    timestamp: int
+    event_hash: str
+
+
+def transform_event_for_linking_payload(replay_id: str, event: BaseEvent) -> EventLinkKafkaMessage:
+    def _make_json_binary_payload() -> EventLinkPayload:
+        return {
+            "type": "event_link",
+            "replay_id": replay_id,
+            "error_id": event.event_id,
+            "timestamp": int(event.datetime.timestamp()),
+            "event_hash": md5((replay_id + event.event_id).encode("utf-8")).hexdigest(),
```

Comment:
The dependency of replay-id + event-id is not necessary since the replay-id is already part of the primary key.  If `event.event_id` is a UUID you can pass it as the event_hash.  Otherwise it can be md5 hashed as it is now.